### PR TITLE
Handle none values more gracefully for FFMC calculations

### PR DESCRIPTION
**Pull request by conbrad**
We weren't handling None to NULL conversions in our calls to CFFDRS' R function, `fine_fuel_moisture_code`, `drought_code` and `drought_moisture_code`.
This PR explicitly converts parameters to NULL if they are None similar to other functions, as well as checks for different types of failed result output from the R library.

Note: the windspeed is mandatory to fail fast on because the R library throws an exception if handed a NULL.
# Test Links:
[Landing Page](https://wps-pr-3629-e1e498-dev.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-3629-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[Percentile Calculator](https://wps-pr-3629-e1e498-dev.apps.silver.devops.gov.bc.ca/percentile-calculator)
[MoreCast](https://wps-pr-3629-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-3629-e1e498-dev.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-3629-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-3629-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-3629-e1e498-dev.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-3629-e1e498-dev.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: api/app/fire_behaviour/cffdrs.py [hunk 1]
```diff
@@ -545,12 +545,25 @@ def fine_fuel_moisture_code(ffmc: float, temperature: float, relative_humidity:
...
         ffmc = NULL
+    if temperature is None:
+        temperature = NULL
+    if relative_humidity is None:
+        relative_humidity = NULL
+    if precipitation is None:
+        precipitation = NULL
+    if wind_speed is None:
+        logger.error("Failed to calculate ffmc")
+        return None
```
Comment (brettedw) [inline api/app/fire_behaviour/cffdrs.py:556]:
```suggestion
        wind_speed = NULL
```

File: api/app/fire_behaviour/cffdrs.py [hunk 2]
```diff
@@ -545,12 +545,25 @@ def fine_fuel_moisture_code(ffmc: float, temperature: float, relative_humidity:
 
     if ffmc is None:
         ffmc = NULL
```
Comment (brettedw) [inline api/app/fire_behaviour/cffdrs.py:547]:
```suggestion
        logger.error("Failed to calculate FFMC; initial FFMC is required.")
        return None
```

Comment (brettedw) [reply api/app/fire_behaviour/cffdrs.py:547]: I guess we can't calculate any of the indices if any provided weather input is None/NULL. Maybe we should just do an `any` check to see if any of them are `None` and avoid calling the R library at all? Either way works I suppose

Comment (dgboss) [reply api/app/fire_behaviour/cffdrs.py:547]: These R functions are so inconsistent. Some will throw an error if not given an initial value for the FWI of interest, others simply return None (or whatever the equivalent is). Some functions allow NULL for a weather variable and others don't. 

I'll add this change but the leave the rest as is. Hopefully we'll get rid of this code in the next few months.

File: api/app/fire_behaviour/cffdrs.py (hunk 1 above)
Comment (dgboss) [reply api/app/fire_behaviour/cffdrs.py:556]: The R ffmc function will fail if passed a NULL windspeed, so we have to short circuit and return None here. I've added a code comment to make this clear.

**Comment by dgboss** [comment]
Clsoing to save resources

**Review by brettedw — APPROVED** [review]
Looks good!

**Review by dgboss — APPROVED** [review]

**Pull request merged**